cluster-autoscaler/simulator/delta_cluster_snapshot.go

```diff
@@ -168,20 +164,11 @@ func (data *internalDeltaSnapshotData) addNodeInfo(nodeInfo *schedulernodeinfo.N
 		data.nodeInfoList = append(data.nodeInfoList, nodeInfo)
 	}
 
-	if len(nodeInfo.Pods()) > 0 {
-		data.clearPodCaches()
-	}
```

Comment:
I think we still need to drop havePodsWithAffinity here.

Reply:
cc: @aleksandra-malinowska

Reply:
Ohh right, that's my bad!